File: CareConnect/CareConnect.Models/Responses/Permission.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace CareConnect.Models.Responses
{
    public class Permission
    {
        public int PermissionId { get; set; }
        
        public string Name { get; set; } = null!;
        
        public string Category { get; set; } = null!;

        public string Action { get; set; } = null!;
        
        public string Resource { get; set; } = null!;
    }
}

File: UI/careconnect_admin/lib/models/responses/permission.dart
import 'package:json_annotation/json_annotation.dart';

part 'permission.g.dart';

@JsonSerializable()
class Permission {
  final int permissionId;
  final String name;
  final String category;
  final String action;
  final String resource;

  Permission({
    required this.permissionId,
    required this.name,
    required this.category,
    required this.action,
    required this.resource,
  });

  factory Permission.fromJson(Map<String, dynamic> json) =>
      _$PermissionFromJson(json);

  Map<String, dynamic> toJson() => _$PermissionToJson(this);
}
